### DISCUSSION TITLE
Topping E30 clicks

### DISCUSSION THREAD

**Discussion by koowat** [rern/rAudio#296] [Bug Reports]
Hi everyone,

New convert to rAudio-1 running the following:
rAudio 1 • 20210330
5.10.25-4-ARCH armv7l
Raspberry Pi 3 Model B Rev 1.2
BCM2837 • 962MiB
4 Cortex-A53 @ 1.2GHz

I am getting audio clicks on DSD64 dsf files despite using a USB memory key (so its not wifi issue). Also the same setup using a TEAC UD-501 has NO audio clicks so I'm wondering if its some MPD implementation. Any ideas?

**Comment by Roland-B** [comment]
Afternoon,

Using Topping E30 with :

rAudio 1 • 20210403
5.10.25-5-ARCH armv7l
Raspberry Pi 4 Model B Rev 1.2
BCM2711 • 3.8GiB
4 Cortex-A72 @ 1.5GHz

Have no clicks playing [Demo Track](http://download.oppodigital.com/hra/davidelias/08%20-%20David%20Elias%20-%20Crossing%20-%20Morning%20Light%20Western%20Town%20%28DSD64%202.0%29.dsf) at all via playing from NAS using ethernet cable to the Pi.

Roland B

**Comment by klassikmann** [comment]
@koowat 

> I am getting audio clicks on DSD64 dsf ...

Question:
Do you use DSF files from a music provider as the demo track of @Roland-B or e.g. DSFs from [https://www.nativedsd.com/](https://www.nativedsd.com/). Or "self-made" DSFs produced by sacd_ripper/sacd_extract? I ask because people are complaining about pops/clicks in DSFs generated by the "original" sacd_extract. In that case I recommend to use sacd_extract from [https://github.com/EuFlo/sacd-ripper](url).
There is a new "-z"-Option:
`-z, --dsf-nopad                 : Do not zero pad DSF`

klassikmann

**Reply by koowat** [reply]
"original" klassikmann - and wonderful tip! I use -s on the original. Was not aware this was an issue. But just curious why too - the Teac UD-501 has no issue (which led me to a mental rabbit hole of whether theres better jitter correction on the Teac).

**Reply by koowat** [reply]
given the -z is zero pad DSF at end of track, that's not the issue - its happening within track and not during track transition. Happens less frequent on a very old Schitt Loki (that one is only DSD64) but Teac UD-501 has been flawless.

**Reply by klassikmann** [reply]
@koowat I guess there is a problem with the driver/firmware of the Topping E30. Perhaps you'll find advice at 
[https://www.audiosciencereview.com/forum/index.php?threads/topping-e30-dac-review.12119/](url)
But beware: It is a 218 pages thread.

btw.: I started with a Teac UD-501, a remarkeble good product. My oldest contributions to the RuneAudio forum are from this time. But since some years I have no need anymore for a DAC, because I replaced the AMP (Denon PMA-S 10 II), the speakers (JBL Ti 5000) and the UD-501 by digitally connected active speakers (Genelec 8351) and a Digital-Out (HifiBerry Digi+ Pro).

klassikmann

**Reply by koowat** [reply]
Good point klassikmann and I'm from same era too! I should be more empirical/objective in evaluating but to me the Teac UD-501 still sounds better than E30 (which would be more versatile as it is a digital preamp/has DSD256)...makes me wonder whether my listening justifies Teac UD-505 (must have borrowed some R&D from Esoteric line)....unfortunately the 2011 serial number I have isnt part of any firmware update (makes me wonder if the more recent productions are plagued by same USB issues)
